File: LearningDSA/array3.cpp
#include<iostream>
using namespace std;

int arrDisplayer(int x[],int y){
    cout<<"[";
    for (int i = 0; i < y-1; i++)
    {
        cout<<x[i]<<",";
    }
    cout<<x[y-1];
    cout<<"]\n\n";
    return 0;
}

int duplicateRemover(int x, int y[]){
    int newArr[x] = {};
    int arrIndex = 0;
    for (int i = 0; i < x; i++)
    {
        int cnt = 0;
        for (int j = i; j < x; j++)
        {
            if (y[i]==y[j])
            {
                cnt++;
            }                        
        }
        int lenNewArr = sizeof(newArr);
        cout<<lenNewArr<<endl;
        if (cnt>1 && newArr[lenNewArr-1]!=y[i])
        {
            newArr[arrIndex] = y[i];
            arrIndex++;
        }         
    }
    cout << "Array with duplicates removed:"<<endl;
    arrDisplayer(newArr,arrIndex+1);
    return 0;
}

int main()
{
    int arr[10] = {1,1,4,6,6,6,9,9,45,67};
    int arrLen = sizeof(arr);
    duplicateRemover(arrLen,arr);
    
    return 0;
}

File: LearningDSA/array3.py
def duplicateRemover(arr):
    new_arr = []
    for i in range(0,len(arr)):
        count=0
        for j in range(i, len(arr)):
            if arr[i]==arr[j]:
                count+=1
        if count>=1 and len(new_arr)==0:
            new_arr.append(arr[i])
        elif count>=1 and new_arr[-1]!=arr[i]:
            new_arr.append(arr[i])
    print(new_arr)

list = [1,1,4,6,6,6,9,9,31,31,31,31,45,67]
duplicateRemover(list)